Add tests for Meeting commands #95

lilyium opened the pull request:
Fixes #95 

Add tests for:
- [x] AddMeetingCommand
- [x] DeleteMeetingCommand
- [x] AddMeetingCommandParser
- [x] DeleteMeetingCommandParser 

Proposed commit message:
```
Add tests for Meeting commands in the logic component

Let's add tests for 
* AddMeetingCommand
* DeleteMeetingCommand
* AddMeetingCommandParser
* DeleteMeetingCommandParser

Note: Tests related to Meeting in the model and storage component have yet to be added
```

lilyium committed:
Add MeetingBuilder class to help with building Meeting objects

Make date and time fields private in Meeting class
Add getters to Meeting class

lilyium committed:
Add tests for AddMeetingCommand

Rename TypicalPersons to TypicalAddressBook.
Add Meeting objects to TypicalAddressBook that are to be used in tests.

lilyium committed:
Update test for ListMeetingsCommand

Rename ListMeetingsTest to ListMeetingsCommandTest.

lilyium committed:
Fix import order

lilyium committed:
Merge branch 'master' into add-test-meeting-command

lilyium committed:
Add tests for DeleteMeetingCommand

lilyium committed:
Update toString method for DeleteMeetingCommand

lilyium committed:
Add a manually added meeting to TypicalAddressBook and CommandTestUtil

Rename fields in MeetingBuilder to split STARTTIME and ENDTIME into
START_TIME and END_TIME respectively

lilyium committed:
Add tests for AddMeetingCommandParser

lilyium committed:
Add tests for DeleteMeetingCommandParser

lilyium committed:
Remove unused imports

leewaikin19 commented on src/main/java/syncsquad/teamsync/model/meeting/Meeting.java:
```diff
@@ -32,6 +32,18 @@ public Meeting(LocalDate date, LocalTime startTime, LocalTime endTime) {
         this.endTime = endTime;
     }
 
+    public LocalDate getDate() {
```
Oops it seems like I also added these lines

leewaikin19 reviewed (approved):
LGTM

Kannan171 commented on src/test/java/syncsquad/teamsync/logic/commands/DeleteMeetingTest.java:
```diff
@@ -0,0 +1,79 @@
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteMeetingCommand}.
+ */
+public class DeleteMeetingTest {
```
```suggestion
public class DeleteMeetingCommandTest {
```

lilyium replied:
Java file and class name renamed to `DeleteMeetingCommandTest` for consistency

leewaikin19 commented on src/test/java/syncsquad/teamsync/logic/commands/AddMeetingCommandTest.java:
```diff
@@ -0,0 +1,225 @@
+        public void deleteMeeting(Meeting meeting) {
+            throw new AssertionError("This method should not be called.");
+        }
+
```
Have added a new method to the Model, so a new override method needs to be added here.

Kannan171 commented on src/test/java/syncsquad/teamsync/logic/commands/DeleteMeetingTest.java:
```diff
@@ -0,0 +1,79 @@
+
+    @Test
+    public void execute_validIndexList_success() {
+        Meeting meetingToDelete = model.getMeetingList().get(INDEX_FIRST_PERSON.getZeroBased());
```
```suggestion
        Meeting meetingToDelete = model.getMeetingList().get(INDEX_FIRST.getZeroBased());
```

lilyium replied:
Renamed `INDEX_FIRST_PERSON`, `INDEX_SECOND_PERSON` and `INDEX_THIRD_PERSON` to `INDEX_FIRST`, `INDEX_SECOND` and `INDEX_THIRD` as Index is used for both Person and Meeting lists.

Kannan171 reviewed (approved):
LGTM! Just a few nits to fix.

lilyium committed:
Rename DeleteMeetingTest to DeleteMeetingCommandTest

Rename INDEX_FIRST_PERSON, INDEX_SECOND_PERSON, INDEX_THIRD_PERSON to
INDEX_FIRST, INDEX_SECOND, INDEX_THIRD as the Index is used for both
Person and Meeting
